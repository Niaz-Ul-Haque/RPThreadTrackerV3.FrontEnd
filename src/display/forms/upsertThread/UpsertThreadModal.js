// #region imports
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Modal, ModalHeader, ModalBody, ModalFooter, Button } from 'reactstrap';
import { AvForm } from 'availity-reactstrap-validation';
import TooltipForm from '../TooltipForm';
import UpsertThreadForm from './UpsertThreadForm';
import validator from './_validator';
import formData from './_formData';
// #endregion imports

const propTypes = {
	isUpsertThreadModalOpen: PropTypes.bool.isRequired,
	submitUpsertThread: PropTypes.func.isRequired,
	closeUpsertThreadModal: PropTypes.func.isRequired,
	threadToEdit: PropTypes.shape({}).isRequired,
	characters: PropTypes.arrayOf(PropTypes.shape({})).isRequired
};

class UpsertThreadModal extends Component {
	constructor(props) {
		super(props);
		this.selectCharacter = this.selectCharacter.bind(this);
		this.handleInputChange = this.handleInputChange.bind(this);
		this.handleTagAdded = this.handleTagAdded.bind(this);
		this.handleTagRemoved = this.handleTagRemoved.bind(this);
		this.getTagValues = this.getTagValues.bind(this);
		this.state = {
			threadToEdit: props.threadToEdit
		};
	}

	getInitialThreadData(nextProps) {
		const { threadToEdit, characters } = nextProps;
		if (!threadToEdit.characterId && characters.length) {
			threadToEdit.characterId = characters[0].characterId;
		}
		return threadToEdit;
	}

	selectCharacter(characterId) {
		if (this.state.threadToEdit.characterId !== characterId) {
			this.setState({ threadToEdit: { ...this.state.threadToEdit, characterId } });
		}
	}

	handleInputChange(event) {
		const { target } = event;
		const value = target.type === 'checkbox' ? target.checked : target.value;
		const { name } = target;
		this.setState({
			threadToEdit: Object.assign({}, this.state.threadToEdit, {
				[name]: value
			})
		});
	}

	handleTagAdded(tagValue) {
		const currentTags = this.state.threadToEdit.threadTags;
		const newTag = { tagText: tagValue };
		const newTags = currentTags.concat(newTag);
		this.setState({
			threadToEdit: Object.assign({}, this.state.threadToEdit, {
				threadTags: newTags
			})
		});
	}

	handleTagRemoved(tagValue) {
		const currentTags = this.state.threadToEdit.threadTags;
		const newTags = currentTags.filter(tag => tag.tagText !== tagValue);
		this.setState({
			threadToEdit: Object.assign({}, this.state.threadToEdit, {
				threadTags: newTags
			})
		});
	}

	getTagValues() {
		if (!this.state.threadToEdit.threadTags) {
			return [];
		}
		return this.state.threadToEdit.threadTags.map(t => t.tagText);
	}

	render() {
		const {
			isUpsertThreadModalOpen,
			submitUpsertThread,
			closeUpsertThreadModal,
			threadToEdit,
			characters
		} = this.props;
		if (!threadToEdit) {
			return (
				<div />
			);
		}
		return (
			<Modal isOpen={isUpsertThreadModalOpen} toggle={closeUpsertThreadModal} backdrop>
				<AvForm onValidSubmit={() => submitUpsertThread(this.state.threadToEdit)}>
					<ModalHeader toggle={closeUpsertThreadModal}>
						{threadToEdit && threadToEdit.threadId ? 'Edit Thread' : 'Add New Thread'}
					</ModalHeader>
					<ModalBody>
						<TooltipForm
							Renderable={UpsertThreadForm}
							threadToEdit={threadToEdit}
							validator={validator}
							formData={formData}
							characters={characters}
							selectCharacter={this.selectCharacter}
							handleInputChange={this.handleInputChange}
							handleTagAdded={this.handleTagAdded}
							handleTagRemoved={this.handleTagRemoved}
							tagValues={this.getTagValues()}
						/>
					</ModalBody>
					<ModalFooter>
						<Button color="primary">{threadToEdit.threadId ? 'Edit Thread' : 'Add Thread'}</Button>{' '}
						<Button color="secondary" onClick={closeUpsertThreadModal}>Cancel</Button>
					</ModalFooter>
				</AvForm>
			</Modal>
		);
	}
}

UpsertThreadModal.propTypes = propTypes;
export default UpsertThreadModal;
