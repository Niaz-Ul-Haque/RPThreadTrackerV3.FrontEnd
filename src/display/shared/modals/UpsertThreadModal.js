import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Modal, ModalHeader, ModalBody, ModalFooter, Col, Row, Form, FormGroup, Label, Input, FormText, Button } from 'reactstrap';
import { AvForm, AvField } from 'availity-reactstrap-validation';
import CharacterSelect from '../CharacterSelect';
import { upsertThreadValidator } from '../../../infrastructure/validators';

const propTypes = {
	isUpsertThreadModalOpen: PropTypes.bool.isRequired,
	submitUpsertThread: PropTypes.func.isRequired,
	closeUpsertThreadModal: PropTypes.func.isRequired,
	threadToEdit: PropTypes.shape({}).isRequired,
	characters: PropTypes.shape({}).isRequired
};

class UpsertThreadModal extends Component {
	constructor(props) {
		super();
		this.selectCharacter = this.selectCharacter.bind(this);
		this.handleInputChange = this.handleInputChange.bind(this);
		this.state = {
			threadToEdit: props.threadToEdit
		};
	}

	selectCharacter(characterId) {
		this.setState({ threadToEdit: { ...this.state.threadToEdit, characterId } });
	}

	handleInputChange(event) {
		const { target } = event;
		const value = target.type === 'checkbox' ? target.checked : target.value;
		const { name } = target;
		this.setState({
			threadToEdit: Object.assign({}, this.state.threadToEdit, {
				[name]: value
			})
		});
	}
	render() {
		const {
			isUpsertThreadModalOpen,
			submitUpsertThread,
			closeUpsertThreadModal,
			threadToEdit,
			characters
		} = this.props;
		if (!threadToEdit) {
			return (
				<div />
			);
		}
		return (
			<Modal isOpen={isUpsertThreadModalOpen} toggle={closeUpsertThreadModal} backdrop>
				<AvForm onValidSubmit={() => submitUpsertThread(this.state.threadToEdit)}>
					<ModalHeader toggle={closeUpsertThreadModal}>{threadToEdit.id ? 'Edit Thread' : 'Add New Thread'}</ModalHeader>
					<ModalBody>
						<FormGroup row>
							<Col>
								<CharacterSelect
									characters={characters}
									selectedCharacterId={this.state.threadToEdit.characterId}
									onSelectCharacter={this.selectCharacter}
									includeNullValue={false}
								/>
							</Col>
						</FormGroup>
						<Row>
							<Col>
								<AvField
									name="userTitle"
									placeholder="Thread Title"
									label="Thread Title"
									type="text"
									onChange={this.handleInputChange}
									validate={upsertThreadValidator.userTitle}
									helpMessage="This can be anything you like!"
								/>
							</Col>
						</Row>
						<Row>
							<Col>
								<AvField
									name="postId"
									placeholder="Post ID"
									label="Post ID"
									type="text"
									onChange={this.handleInputChange}
									validate={upsertThreadValidator.postId}
									helpMessage={[
										'This must be a post from your blog. The post ID is the ',
										'part of the URL after ".tumblr.com/post/". For instance, ',
										'if the post is at the URL ',
										<strong>http://myawesomeblog.tumblr.com/post/12345</strong>,
										', you would enter ',
										<strong>12345</strong>,
										' in this field.']}
								/>
							</Col>
						</Row>
						<Row>
							<Col>
								<AvField
									name="partnerUrlIdentifier"
									placeholder="Partner Url Identifier"
									label="Partner URL Identifier"
									type="text"
									onChange={this.handleInputChange}
									validate={upsertThreadValidator.partnerUrlIdentifier}
									helpMessage={[
										'This will be the part of your partner\'s URL before ',
										'".tumblr.com". For instance, if their URL is',
										<strong>http://myawesomeblog.tumblr.com</strong>,
										', you would enter',
										<strong>myawesomeblog</strong>,
										' in this field.']}
								/>
							</Col>
						</Row>
					</ModalBody>
					<ModalFooter>
						<Button color="primary">Add Thread</Button>{' '}
						<Button color="secondary" onClick={closeUpsertThreadModal}>Cancel</Button>
					</ModalFooter>
				</AvForm>
			</Modal>
		);
	}
}

UpsertThreadModal.propTypes = propTypes;
export default UpsertThreadModal;
