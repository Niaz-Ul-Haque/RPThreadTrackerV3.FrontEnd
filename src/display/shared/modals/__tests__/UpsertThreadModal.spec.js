// #region imports
import React from 'react';
import { shallow } from 'enzyme';
import { getSpecWrapper } from '../../../../../config/tests/helpers.unit';
import UpsertThreadModal from '../UpsertThreadModal';
// #endregion imports

// #region mocks
jest.mock('../../../forms/upsert-thread/UpsertThreadForm', () => 'UpsertThreadForm');
jest.mock('../../../forms/TooltipForm', () => 'TooltipForm');
jest.mock('../../styled/Modal', () => 'Modal');
// #endregion mocks

const createTestProps = propOverrides => ({
	isUpsertThreadModalOpen: true,
	submitUpsertThread: jest.fn(),
	closeUpsertThreadModal: jest.fn(),
	threadToEdit: { threadId: 1 },
	characters: [{ characterId: 1 }, { characterId: 2 }],
	...propOverrides
});

describe('rendering', () => {
	describe('snapshots', () => {
		it('should render valid snapshot', () => {
			const props = createTestProps();
			const jsx = (<UpsertThreadModal {...props} />);
			const element = shallow(jsx);
			expect(element).toMatchSnapshot();
		});
		it('should render valid snapshot if thread is new', () => {
			const props = createTestProps({ threadToEdit: {} });
			const jsx = (<UpsertThreadModal {...props} />);
			const element = shallow(jsx);
			expect(element).toMatchSnapshot();
		});
		it('should render valid snapshot if thread has tags', () => {
			const threadToEdit = { threadId: 1, threadTags: [{ tagText: 'tag1' }, { tagText: 'tag2' }] };
			const props = createTestProps({ threadToEdit });
			const jsx = (<UpsertThreadModal {...props} />);
			const element = shallow(jsx);
			expect(element).toMatchSnapshot();
			const form = element.find('TooltipForm');
			expect(form.prop('tagValues')).toHaveLength(2);
		});
	});
});

describe('behavior', () => {
	describe('submitUpsertThread', () => {
		it('should be called when form is submitted', () => {
			const submitUpsertThread = jest.fn();
			const threadToEdit = { threadId: 2 };
			const props = createTestProps({ submitUpsertThread, threadToEdit });
			const jsx = (<UpsertThreadModal {...props} />);
			const element = shallow(jsx);
			const form = getSpecWrapper(element, 'upsert-thread-modal-form');
			form.prop('onValidSubmit')();
			expect(submitUpsertThread).toHaveBeenCalledTimes(1);
			expect(submitUpsertThread).toHaveBeenLastCalledWith(threadToEdit);
		});
	});
	describe('handleInputChange', () => {
		it('should handle text field update', () => {
			const submitUpsertThread = jest.fn();
			const event = { target: { type: 'text', name: 'userTitle', value: 'My Thread' } };
			const props = createTestProps({ submitUpsertThread });
			const jsx = (<UpsertThreadModal {...props} />);
			const element = shallow(jsx);
			element.instance().handleInputChange(event);
			element.update();
			const form = getSpecWrapper(element, 'upsert-thread-modal-form');
			form.prop('onValidSubmit')();
			expect(submitUpsertThread).toHaveBeenLastCalledWith({ threadId: 1, userTitle: 'My Thread' });
		});
		it('should handle checkbox field input', () => {
			const submitUpsertThread = jest.fn();
			const event = { target: { type: 'checkbox', name: 'isArchived', checked: true } };
			const props = createTestProps({ submitUpsertThread });
			const jsx = (<UpsertThreadModal {...props} />);
			const element = shallow(jsx);
			element.instance().handleInputChange(event);
			element.update();
			const form = getSpecWrapper(element, 'upsert-thread-modal-form');
			form.prop('onValidSubmit')();
			expect(submitUpsertThread).toHaveBeenLastCalledWith({ threadId: 1, isArchived: true });
		});
	});
	describe('componentWillReceiveProps', () => {
		it('should set thread to edit', () => {
			const submitUpsertThread = jest.fn();
			const threadToEdit = { threadId: 2, characterId: 2 };
			const props = createTestProps({ submitUpsertThread });
			const jsx = (<UpsertThreadModal {...props} />);
			const element = shallow(jsx);
			element.setProps({ threadToEdit });
			const form = getSpecWrapper(element, 'upsert-thread-modal-form');
			form.prop('onValidSubmit')();
			expect(submitUpsertThread).toHaveBeenLastCalledWith(threadToEdit);
		});
		it('should set thread to edit with null character ID if character ID not present on thread', () => {
			const submitUpsertThread = jest.fn();
			const threadToEdit = { threadId: 2 };
			const props = createTestProps({ submitUpsertThread });
			const jsx = (<UpsertThreadModal {...props} />);
			const element = shallow(jsx);
			element.setProps({ threadToEdit });
			const form = getSpecWrapper(element, 'upsert-thread-modal-form');
			form.prop('onValidSubmit')();
			expect(submitUpsertThread).toHaveBeenLastCalledWith(threadToEdit);
		});
	});
	describe('closeUpsertThreadModal', () => {
		it('should be triggered when modal is toggled', () => {
			const closeUpsertThreadModal = jest.fn();
			const props = createTestProps({ closeUpsertThreadModal });
			const jsx = (<UpsertThreadModal {...props} />);
			const element = shallow(jsx);
			const modal = getSpecWrapper(element, 'upsert-thread-modal');
			modal.prop('toggle')();
			expect(closeUpsertThreadModal).toHaveBeenCalledTimes(1);
		});
		it('should be triggered when modal header is toggled', () => {
			const closeUpsertThreadModal = jest.fn();
			const props = createTestProps({ closeUpsertThreadModal });
			const jsx = (<UpsertThreadModal {...props} />);
			const element = shallow(jsx);
			const header = getSpecWrapper(element, 'upsert-thread-modal-header');
			header.prop('toggle')();
			expect(closeUpsertThreadModal).toHaveBeenCalledTimes(1);
		});
		it('should be triggered when close button is clicked', () => {
			const closeUpsertThreadModal = jest.fn();
			const props = createTestProps({ closeUpsertThreadModal });
			const jsx = (<UpsertThreadModal {...props} />);
			const element = shallow(jsx);
			const button = getSpecWrapper(element, 'upsert-thread-modal-close-button');
			button.simulate('click');
			expect(closeUpsertThreadModal).toHaveBeenCalledTimes(1);
		});
	});
	describe('handleTagRemoved', () => {
		it('should remove tag', () => {
			const submitUpsertThread = jest.fn();
			const threadToEdit = { threadId: 1, threadTags: [{ tagText: 'Test Tag 1' }, { tagText: 'Test Tag 2' }] };
			const props = createTestProps({ threadToEdit, submitUpsertThread });
			const jsx = (<UpsertThreadModal {...props} />);
			const element = shallow(jsx);
			element.instance().handleTagRemoved('Test Tag 1');
			element.update();
			const form = getSpecWrapper(element, 'upsert-thread-modal-form');
			form.prop('onValidSubmit')();
			expect(submitUpsertThread).toHaveBeenLastCalledWith({ threadId: 1, threadTags: [{ tagText: 'Test Tag 2' }] });
		});
		it('should instantiate threadTags before attempting remove if it does not exist', () => {
			const submitUpsertThread = jest.fn();
			const threadToEdit = { threadId: 1 };
			const props = createTestProps({ threadToEdit, submitUpsertThread });
			const jsx = (<UpsertThreadModal {...props} />);
			const element = shallow(jsx);
			element.instance().handleTagRemoved('Test Tag 1');
			element.update();
			const form = getSpecWrapper(element, 'upsert-thread-modal-form');
			form.prop('onValidSubmit')();
			expect(submitUpsertThread).toHaveBeenLastCalledWith({ threadId: 1, threadTags: [] });
		});
		it('should do nothing if tag does not exist in threadTags', () => {
			const submitUpsertThread = jest.fn();
			const threadToEdit = { threadId: 1, threadTags: [{ tagText: 'Test Tag 1' }, { tagText: 'Test Tag 2' }] };
			const props = createTestProps({ threadToEdit, submitUpsertThread });
			const jsx = (<UpsertThreadModal {...props} />);
			const element = shallow(jsx);
			element.instance().handleTagRemoved('Test Tag 3');
			const form = getSpecWrapper(element, 'upsert-thread-modal-form');
			form.prop('onValidSubmit')();
			expect(submitUpsertThread).toHaveBeenLastCalledWith({ threadId: 1, threadTags: [{ tagText: 'Test Tag 1' }, { tagText: 'Test Tag 2' }] });
		});
	});
	describe('handleTagAdded', () => {
		it('should add tag', () => {
			const submitUpsertThread = jest.fn();
			const threadToEdit = { threadId: 1, threadTags: [{ tagText: 'Test Tag 1' }, { tagText: 'Test Tag 2' }] };
			const props = createTestProps({ threadToEdit, submitUpsertThread });
			const jsx = (<UpsertThreadModal {...props} />);
			const element = shallow(jsx);
			element.instance().handleTagAdded('Test Tag 3');
			element.update();
			const form = getSpecWrapper(element, 'upsert-thread-modal-form');
			form.prop('onValidSubmit')();
			expect(submitUpsertThread).toHaveBeenLastCalledWith({ threadId: 1, threadTags: [{ tagText: 'Test Tag 1' }, { tagText: 'Test Tag 2' }, { tagText: 'Test Tag 3' }] });
		});
		it('should instantiate threadTags before adding if array does not exist', () => {
			const submitUpsertThread = jest.fn();
			const threadToEdit = { threadId: 1 };
			const props = createTestProps({ threadToEdit, submitUpsertThread });
			const jsx = (<UpsertThreadModal {...props} />);
			const element = shallow(jsx);
			element.instance().handleTagAdded('Test Tag 1');
			element.update();
			const form = getSpecWrapper(element, 'upsert-thread-modal-form');
			form.prop('onValidSubmit')();
			expect(submitUpsertThread).toHaveBeenLastCalledWith({ threadId: 1, threadTags: [{ tagText: 'Test Tag 1' }] });
		});
		it('should do nothing if the tag already exists in threadTags', () => {
			const submitUpsertThread = jest.fn();
			const threadToEdit = { threadId: 1, threadTags: [{ tagText: 'Test Tag 1' }, { tagText: 'Test Tag 2' }] };
			const props = createTestProps({ threadToEdit, submitUpsertThread });
			const jsx = (<UpsertThreadModal {...props} />);
			const element = shallow(jsx);
			element.instance().handleTagAdded('Test Tag 2');
			const form = getSpecWrapper(element, 'upsert-thread-modal-form');
			form.prop('onValidSubmit')();
			expect(submitUpsertThread).toHaveBeenLastCalledWith({ threadId: 1, threadTags: [{ tagText: 'Test Tag 1' }, { tagText: 'Test Tag 2' }] });
		});
	});
	describe('selectCharacter', () => {
		it('should set character ID', () => {
			const event = { target: { value: 5 } };
			const submitUpsertThread = jest.fn();
			const threadToEdit = { threadId: 1, characterId: 2 };
			const props = createTestProps({ threadToEdit, submitUpsertThread });
			const jsx = (<UpsertThreadModal {...props} />);
			const element = shallow(jsx);
			element.instance().selectCharacter(event);
			element.update();
			const form = getSpecWrapper(element, 'upsert-thread-modal-form');
			form.prop('onValidSubmit')();
			expect(submitUpsertThread).toHaveBeenLastCalledWith({ threadId: 1, characterId: 5 });
		});
		it('should do nothing if ID has not changed', () => {
			const event = { target: { value: 1 } };
			const submitUpsertThread = jest.fn();
			const threadToEdit = { threadId: 1, characterId: 1 };
			const props = createTestProps({ threadToEdit, submitUpsertThread });
			const jsx = (<UpsertThreadModal {...props} />);
			const element = shallow(jsx);
			element.instance().selectCharacter(event);
			const form = getSpecWrapper(element, 'upsert-thread-modal-form');
			form.prop('onValidSubmit')();
			expect(submitUpsertThread).toHaveBeenLastCalledWith({ threadId: 1, characterId: 1 });
		});
	});
});
