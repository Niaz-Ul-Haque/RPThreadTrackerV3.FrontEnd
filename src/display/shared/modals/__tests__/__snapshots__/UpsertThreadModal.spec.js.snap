// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering snapshots should render valid snapshot 1`] = `
<Styled(Modal)
  backdrop={true}
  data-spec="upsert-thread-modal"
  isOpen={true}
  toggle={[MockFunction]}
>
  <AvForm
    data-spec="upsert-thread-modal-form"
    method="get"
    model={Object {}}
    onInvalidSubmit={[Function]}
    onKeyDown={[Function]}
    onSubmit={[Function]}
    onValidSubmit={[Function]}
    tag={[Function]}
    validationEvent={
      Array [
        "onChange",
        "onInput",
      ]
    }
  >
    <ModalHeader
      charCode={215}
      closeAriaLabel="Close"
      data-spec="upsert-thread-modal-header"
      tag="h5"
      toggle={[MockFunction]}
      wrapTag="div"
    >
      Edit Thread
    </ModalHeader>
    <ModalBody
      tag="div"
    >
      <TooltipForm
        Renderable="UpsertThreadForm"
        characters={
          Array [
            Object {
              "characterId": 1,
            },
            Object {
              "characterId": 2,
            },
          ]
        }
        handleInputChange={[Function]}
        handleTagAdded={[Function]}
        handleTagRemoved={[Function]}
        selectCharacter={[Function]}
        tagValues={Array []}
        threadToEdit={
          Object {
            "threadId": 1,
          }
        }
      />
    </ModalBody>
    <ModalFooter
      tag="div"
    >
      <Button
        color="primary"
        tag="button"
      >
        Edit Thread
      </Button>
       
      <Button
        color="secondary"
        data-spec="upsert-thread-modal-close-button"
        onClick={[MockFunction]}
        tag="button"
      >
        Cancel
      </Button>
    </ModalFooter>
  </AvForm>
</Styled(Modal)>
`;

exports[`rendering snapshots should render valid snapshot if thread has tags 1`] = `
<Styled(Modal)
  backdrop={true}
  data-spec="upsert-thread-modal"
  isOpen={true}
  toggle={[MockFunction]}
>
  <AvForm
    data-spec="upsert-thread-modal-form"
    method="get"
    model={Object {}}
    onInvalidSubmit={[Function]}
    onKeyDown={[Function]}
    onSubmit={[Function]}
    onValidSubmit={[Function]}
    tag={[Function]}
    validationEvent={
      Array [
        "onChange",
        "onInput",
      ]
    }
  >
    <ModalHeader
      charCode={215}
      closeAriaLabel="Close"
      data-spec="upsert-thread-modal-header"
      tag="h5"
      toggle={[MockFunction]}
      wrapTag="div"
    >
      Edit Thread
    </ModalHeader>
    <ModalBody
      tag="div"
    >
      <TooltipForm
        Renderable="UpsertThreadForm"
        characters={
          Array [
            Object {
              "characterId": 1,
            },
            Object {
              "characterId": 2,
            },
          ]
        }
        handleInputChange={[Function]}
        handleTagAdded={[Function]}
        handleTagRemoved={[Function]}
        selectCharacter={[Function]}
        tagValues={
          Array [
            "tag1",
            "tag2",
          ]
        }
        threadToEdit={
          Object {
            "threadId": 1,
            "threadTags": Array [
              Object {
                "tagText": "tag1",
              },
              Object {
                "tagText": "tag2",
              },
            ],
          }
        }
      />
    </ModalBody>
    <ModalFooter
      tag="div"
    >
      <Button
        color="primary"
        tag="button"
      >
        Edit Thread
      </Button>
       
      <Button
        color="secondary"
        data-spec="upsert-thread-modal-close-button"
        onClick={[MockFunction]}
        tag="button"
      >
        Cancel
      </Button>
    </ModalFooter>
  </AvForm>
</Styled(Modal)>
`;

exports[`rendering snapshots should render valid snapshot if thread is new 1`] = `
<Styled(Modal)
  backdrop={true}
  data-spec="upsert-thread-modal"
  isOpen={true}
  toggle={[MockFunction]}
>
  <AvForm
    data-spec="upsert-thread-modal-form"
    method="get"
    model={Object {}}
    onInvalidSubmit={[Function]}
    onKeyDown={[Function]}
    onSubmit={[Function]}
    onValidSubmit={[Function]}
    tag={[Function]}
    validationEvent={
      Array [
        "onChange",
        "onInput",
      ]
    }
  >
    <ModalHeader
      charCode={215}
      closeAriaLabel="Close"
      data-spec="upsert-thread-modal-header"
      tag="h5"
      toggle={[MockFunction]}
      wrapTag="div"
    >
      Add New Thread
    </ModalHeader>
    <ModalBody
      tag="div"
    >
      <TooltipForm
        Renderable="UpsertThreadForm"
        characters={
          Array [
            Object {
              "characterId": 1,
            },
            Object {
              "characterId": 2,
            },
          ]
        }
        handleInputChange={[Function]}
        handleTagAdded={[Function]}
        handleTagRemoved={[Function]}
        selectCharacter={[Function]}
        tagValues={Array []}
        threadToEdit={Object {}}
      />
    </ModalBody>
    <ModalFooter
      tag="div"
    >
      <Button
        color="primary"
        tag="button"
      >
        Add Thread
      </Button>
       
      <Button
        color="secondary"
        data-spec="upsert-thread-modal-close-button"
        onClick={[MockFunction]}
        tag="button"
      >
        Cancel
      </Button>
    </ModalFooter>
  </AvForm>
</Styled(Modal)>
`;
